UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Common/HealthCheck/AzureWebJobsHealthCheckClient.cs

```diff
@@ -0,0 +1,116 @@
+                foreach (string exchangeSetType in exchangeSetTypes)
+                {
+                    for (int i = 1; i <= GetInstanceCount(exchangeSetType); i++)
+                    {
+                        userNameKey = $"ess-{webHostEnvironment.EnvironmentName}-{exchangeSetType}-{i}-webapp-scm-username";
+                        passwordKey = $"ess-{webHostEnvironment.EnvironmentName}-{exchangeSetType}-{i}-webapp-scm-password";
+                        webJobUri = $"https://ess-{webHostEnvironment.EnvironmentName}-{exchangeSetType}-{i}-webapp.scm.azurewebsites.net/api/continuouswebjobs/ESSFulfilmentWebJob";
+
+                        string userPswd = webJobsAccessKeyProvider.GetWebJobsAccessKey(userNameKey) + ":" + webJobsAccessKeyProvider.GetWebJobsAccessKey(passwordKey);
+                        userPswd = Convert.ToBase64String(Encoding.Default.GetBytes(userPswd));
+                        webJobs.Add(Tuple.Create(userPswd, webJobUri, exchangeSetType, i));
+                    }
+                }
+                var webJobsHealth = CheckAllWebJobsHealth(webJobs);
+                await Task.WhenAll(webJobsHealth);
+
+                return webJobsHealth.Result;
+            }
+            catch (Exception ex)
+            {
+                return HealthCheckResult.Unhealthy($"Azure webjob is unhealthy with error {ex.Message}", new Exception(ex.Message));
+            }
+        }
+
+        private async Task<HealthCheckResult> CheckAllWebJobsHealth(List<Tuple<string, string, string, int>> webJobs)
+        {
+            string webJobStatus = string.Empty;
+            string webJobDetail = string.Empty;
+            foreach (var webJob in webJobs)
+            {
+                using var httpRequestMessage = new HttpRequestMessage(HttpMethod.Get, webJob.Item2);
+                httpClient.DefaultRequestHeaders.Accept.Clear();
+                httpClient.DefaultRequestHeaders.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));
+                httpClient.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Basic", webJob.Item1);
+                var response = await httpClient.SendAsync(httpRequestMessage, HttpCompletionOption.ResponseHeadersRead, CancellationToken.None);
+
+                if (response.StatusCode == HttpStatusCode.OK)
+                {
+                    var webJobDetails = JsonConvert.DeserializeObject<dynamic>(await response.Content.ReadAsStringAsync());
+                    webJobStatus = webJobDetails["status"];
+                    if (webJobStatus != "Running")
+                    {
+                        webJobDetail = $"Webjob ess-{webHostEnvironment.EnvironmentName}-{webJob.Item3}-{webJob.Item4} status is {webJobStatus}";
+                        break;
+                    }
+                }
+                if (webJobStatus != "Running")
+                    break;
```

Comment:
Can we log error and return unhealthy instead of break.
Also log.debug in case of healthy

Reply:
Fixed